src/python_testing/TC_IDM_1_4.py

```diff
@@ -247,8 +247,65 @@ async def steps_3_to_9(self, dummy_value):
         except InteractionModelError:
             asserts.fail("DUT failed with non-path specific error when path specific error was expected")
 
-        # Skipping test until https://github.com/project-chip/connectedhomeip/issues/31434 resolved
-        self.skip_step(9)
+        self.step(9)
+        try:
+            feature_map = await self.read_single_attribute(
+                dev_ctrl,
+                dut_node_id,
+                endpoint=0,
+                attribute=Clusters.GeneralDiagnostics.Attributes.FeatureMap
+            )
+        except InteractionModelError:
+            asserts.fail("DUT failed to respond reading FeatureMap attribute")
+        has_data_model_test_feature = (feature_map & Clusters.GeneralDiagnostics.Bitmaps.Feature.kDataModelTest) != 0
+        asserts.assert_true(has_data_model_test_feature, "DataModelTest Feature is not supported by DUT")
+
+        self.step(10)
+        try:
+            test_event_triggers_enabled = await self.read_single_attribute(
+                dev_ctrl,
+                dut_node_id,
+                endpoint=0,
+                attribute=Clusters.GeneralDiagnostics.Attributes.TestEventTriggersEnabled
+            )
+        except InteractionModelError:
+            asserts.fail("DUT failed to respond reading TestEventTriggersEnabled attribute")
+        asserts.assert_true(test_event_triggers_enabled, "Test Event Triggers are NOT Enabled on DUT")
+
+        self.step(11)
+        asserts.assert_true('PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY' in self.matter_test_config.global_test_params,
+                            "PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY must be included on the command line in "
+                            "the --hex-arg flag as PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY:<key>, "
+                            "e.g. --hex-arg PIXIT.DGGEN.TEST_EVENT_TRIGGER_KEY:000102030405060708090a0b0c0d0e0f")
```

Comment:
Please put this at the very top of the test otherwise it could start even if it will fail.

Reply:
I lifted it as high as I could. Test steps 1 and 2 are still supposed to run on devices where MaxPathsPerInvoke is 1 so PIXIT is not required on those devices